File: c/eu006.c
// project euler (projecteuler.net) problem 6:
// solution by Kevin Retzke (retzkek@gmail.com), November 2009

#include <stdio.h>
#include <stdbool.h>

#include "pe.h"

// SumOfSquares calculates the sum of the squares for all natural 
// numbers from 1 to max.
long SumOfSquares(int max)
{
	long sum = 0;
	for (int i = 1; i <= max; i++) {
		sum += i * i;
	}
	return sum;
}

// SquareOfSum calculates the square of the sum for all natural numbers
// from 1 to max.
long SquareOfSum(int max)
{
	long sum = 0;
	for (int i = 1; i <= max; i++) {
		sum += i;
	}
	return sum * sum;
}

bool test()
{
	return (SquareOfSum(10) - SumOfSquares(10) == 2640);
}

void run()
{
	printf("%ld\n", SquareOfSum(100) - SumOfSquares(100));
}

int main()
{
	return projectEuler(test, run);
}

File: julia/eu006.jl
#!/usr/bin/env julia

# calculates the sum of the squares for all integers from 1 to n
function sum_of_squares(n::Int)
    sum(map(x->x*x, 1:n))
end

# calculates the square of the sum for all integers from 1 to n
function square_of_sum(n::Int)
    s = sum(1:n)
    s*s
end

using Test
@test square_of_sum(10) - sum_of_squares(10) == 2640

println(square_of_sum(100) - sum_of_squares(100))
